lib/ClangImporter/ImportDecl.cpp

```diff
@@ -311,8 +311,13 @@ getSwiftStdlibType(const clang::TypedefNameDecl *D,
     break;
 
   case MappedCTypeKind::VaList:
-    if (ClangTypeSize != ClangCtx.getTypeSize(ClangCtx.VoidPtrTy))
-      return std::make_pair(Type(), "");
+
+    if (ClangTypeSize != ClangCtx.getTypeSize(ClangCtx.VoidPtrTy)) {
+      if (ClangCtx.getTargetInfo().getBuiltinVaListKind() !=
+          clang::TargetInfo::AArch64ABIBuiltinVaList)
+        return std::make_pair(Type(), "");
+    }
```

Comment:
I think that this might be easier to read as:

    if (ClangCtx.getTargetInfo().getBuiltinVaListKind() == clang::TargetInfo::AArch64ABIBuiltinVaList)
      break;
    if (ClangTypeSize == ClangCtx.getTypeSize(ClangCtx.VoidPtrTy))
      break;
    return std::make_pair(Type(), "");

Reply:
Should we just switch over `getBuiltinVaListKind()` instead of considering the size of the type at all?

Reply:
I think the original intent was to make sure it was layout-compatible with Swift.CVaListPointer, which is a simpler proposition, but that would probably work too.

Reply:
@jrose-apple, I think that switching over `getBuiltinVaListKind` accomplishes that and much more explicit.  Just because they they have the same size doesn't necessarily guarantee layout compatibility.